cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletEndpointTest.java

```diff
@@ -0,0 +1,269 @@
+    EasyMock.reset(_mockUUIDGenerator, _mockHttpSession, _mockHttpServletResponse);
+
+    // Transition UserTaskManager state: some tasks will move from ACTIVE to COMPLETED
+    // Resolve futures. Allow those tasks to be moved into completed state
+    getFuture(0).complete(new MockResult());    // Complete 1st requst
+    getFuture(4).complete(new MockResult());    // Complete 5th request
+    getFuture(5).complete(new MockResult());    // Complete 6th request
+    // Update task manager active vs completed state
+    _userTaskManager.checkActiveUserTasks();
+    // Now the UserTaskManager state has changed, so we reload the states
+    UserTaskState userTaskState2 = new UserTaskState(_userTaskManager.getActiveUserTasks(), _userTaskManager.getCompletedUserTasks());
+
+    // Test Case 5: Get all LOAD or REMOVE_BROKER tasks that's completed and with user task id repeatUUID
+    Map<String,  String []> answerQueryParam5 = new HashMap<>();
+    answerQueryParam5.put("param", new String[]{"true"});
+    answerQueryParam5.put("endpoints", new String[]{EndPoint.LOAD.toString() + "," + EndPoint.REMOVE_BROKER.toString()});
+    answerQueryParam5.put("user_task_ids", new String[]{repeatUUID.toString()});
+    answerQueryParam5.put("types", new String[]{UserTaskManager.TaskState.COMPLETED.toString()});
+    HttpServletRequest answerQueryRequest5 = prepareRequest(_mockHttpSession, null, "", EndPoint.USER_TASKS.toString(), answerQueryParam5, GET_METHOD);
+    UserTasksParameters parameters5 = mockUserTasksParameters(answerQueryRequest5);
+    List<UserTaskManager.UserTaskInfo> result5 = userTaskState2.prepareResultList(parameters5);
+    // Test Case 5 result
+    Assert.assertEquals(1, result5.size());
+    EasyMock.reset(_mockUUIDGenerator, _mockHttpSession, _mockHttpServletResponse);
+
+  }
+
+  // Some how we cannot instantiate UserTasksParameters (fail at instantiating LOGGER object), so we mock it.
+  private UserTasksParameters mockUserTasksParameters(HttpServletRequest answerQueryRequest) throws UnsupportedEncodingException {
+    UserTasksParameters parameters = EasyMock.mock(UserTasksParameters.class);
+    EasyMock.expect(parameters.userTaskIds()).andReturn(ParameterUtils.taskIds(answerQueryRequest)).anyTimes();
+    EasyMock.expect(parameters.clientIds()).andReturn(ParameterUtils.clientIds(answerQueryRequest)).anyTimes();
+    EasyMock.expect(parameters.endPoints()).andReturn(ParameterUtils.endPoints(answerQueryRequest)).anyTimes();
+    EasyMock.expect(parameters.endPoint()).andReturn(ParameterUtils.endPoint(answerQueryRequest)).anyTimes();
+    EasyMock.expect(parameters.taskStates()).andReturn(ParameterUtils.types(answerQueryRequest)).anyTimes();
+    EasyMock.expect(parameters.entries()).andReturn(ParameterUtils.entries(answerQueryRequest)).anyTimes();
+
+    EasyMock.replay(parameters);
+    return parameters;
+  }
+
+  private HttpServletRequest prepareTestRequest(HttpSession session, Object userTaskId, Object clientId, Object resource,
```

Comment:
Would `@SuppressWarnings("unchecked")` tag help the compiler understand that this unchecked param is intentional?